From ac205d9ab0ec1cc62d6c08332fc9d95bdff34ff0 Mon Sep 17 00:00:00 2001
From: huang yifeng <huangyifeng@eswincomputing.com>
Date: Mon, 9 Jun 2025 10:32:49 +0800
Subject: [PATCH 387/534] Revert "feat:disable d2d driver && uart noc err log"

This reverts commit f34660b520868e433f0f58dfcbf66b482b7c8427.

Reason for revert: <INSERT REASONING HERE>
---
 .../dts/eswin/eic7702-evb-a1-interleave.dts   |  4 +-
 arch/riscv/boot/dts/eswin/eic7702-evb-a1.dts  |  4 +-
 .../dts/eswin/eic7702-som560-interleave.dtsi  |  4 +-
 arch/riscv/boot/dts/eswin/eic7702-som560.dtsi |  4 +-
 arch/riscv/boot/dts/eswin/eic7702-tb.dts      |  4 +-
 drivers/interconnect/eswin/noc.c              | 56 ++-----------------
 6 files changed, 16 insertions(+), 60 deletions(-)

diff --git a/arch/riscv/boot/dts/eswin/eic7702-evb-a1-interleave.dts b/arch/riscv/boot/dts/eswin/eic7702-evb-a1-interleave.dts
index 6d128b250445..b535e6e6ca54 100644
--- a/arch/riscv/boot/dts/eswin/eic7702-evb-a1-interleave.dts
+++ b/arch/riscv/boot/dts/eswin/eic7702-evb-a1-interleave.dts
@@ -1641,11 +1641,11 @@ &d1_llc_dev{
 };
 
 &d2d {
-	status = "disabled";
+	status = "okay";
 };
 
 &d1_d2d {
-	status = "disabled";
+	status = "okay";
 };
 
 /* die0 GPIO Function Description
diff --git a/arch/riscv/boot/dts/eswin/eic7702-evb-a1.dts b/arch/riscv/boot/dts/eswin/eic7702-evb-a1.dts
index 665b89606a83..fe05cc53ef2f 100644
--- a/arch/riscv/boot/dts/eswin/eic7702-evb-a1.dts
+++ b/arch/riscv/boot/dts/eswin/eic7702-evb-a1.dts
@@ -1607,11 +1607,11 @@ &d1_llc_dev{
 };
 
 &d2d {
-	status = "disabled";
+	status = "okay";
 };
 
 &d1_d2d {
-	status = "disabled";
+	status = "okay";
 };
 
 /* die0 GPIO Function Description
diff --git a/arch/riscv/boot/dts/eswin/eic7702-som560-interleave.dtsi b/arch/riscv/boot/dts/eswin/eic7702-som560-interleave.dtsi
index 65eddea2d49f..2e55ba76ad50 100644
--- a/arch/riscv/boot/dts/eswin/eic7702-som560-interleave.dtsi
+++ b/arch/riscv/boot/dts/eswin/eic7702-som560-interleave.dtsi
@@ -1470,9 +1470,9 @@ &d1_llc_dev{
 };
 
 &d2d {
-	status = "disabled";
+	status = "okay";
 };
 
 &d1_d2d {
-	status = "disabled";
+	status = "okay";
 };
diff --git a/arch/riscv/boot/dts/eswin/eic7702-som560.dtsi b/arch/riscv/boot/dts/eswin/eic7702-som560.dtsi
index c09e33c2180a..5a2cc5c4ec25 100644
--- a/arch/riscv/boot/dts/eswin/eic7702-som560.dtsi
+++ b/arch/riscv/boot/dts/eswin/eic7702-som560.dtsi
@@ -1161,11 +1161,11 @@ &d1_llc_dev{
 };
 
 &d2d {
-	status = "disabled";
+	status = "okay";
 };
 
 &d1_d2d {
-	status = "disabled";
+	status = "okay";
 };
 
 &SOC {
diff --git a/arch/riscv/boot/dts/eswin/eic7702-tb.dts b/arch/riscv/boot/dts/eswin/eic7702-tb.dts
index 148b3240c7a1..59357fbd94a9 100644
--- a/arch/riscv/boot/dts/eswin/eic7702-tb.dts
+++ b/arch/riscv/boot/dts/eswin/eic7702-tb.dts
@@ -1625,11 +1625,11 @@ &d1_llc_dev{
 };
 
 &d2d {
-	status = "disabled";
+	status = "okay";
 };
 
 &d1_d2d {
-	status = "disabled";
+	status = "okay";
 };
 
 /* die0 GPIO Function Description
diff --git a/drivers/interconnect/eswin/noc.c b/drivers/interconnect/eswin/noc.c
index c6d44719db36..cc232396f842 100644
--- a/drivers/interconnect/eswin/noc.c
+++ b/drivers/interconnect/eswin/noc.c
@@ -509,24 +509,6 @@ void stack_dump(void)
 		show_stack(task_list, NULL, KERN_EMERG);
 	}
 }
-/**
- * win2030_noc_get_error - Handle NoC (Network-on-Chip) error interrupts
- * @noc_device: Pointer to the NoC device structure
- *
- * This function is triggered by a NoC error interrupt. It reads all the
- * error log registers, stores them in an error record, and appends the
- * record to the device's internal error queue.
- *
- * Due to a hardware design limitation with the UART module, when both
- * LPCPU and MCPU access the UART concurrently, the NoC may report a
- * spurious error interrupt. These errors are benign and do not lead to
- * hardware malfunction. To reduce unnecessary alarm during normal
- * operation, such specific error cases are logged using dev_dbg().
- *
- * Other errors are logged with dev_err().
- *
- * Return: 0 on success, negative errno on failure (e.g., -ENOMEM)
- */
 static int win2030_noc_get_error(struct win2030_noc_device *noc_device)
 {
 	void __iomem *base = noc_device->hw_base;
@@ -534,58 +516,32 @@ static int win2030_noc_get_error(struct win2030_noc_device *noc_device)
 	unsigned long flags;
 	int i;
 	char buf[2048] = {'\0'};
+
 	struct device *mydev = noc_device->dev;
-	bool use_dbg = false;
 
 	noc_err = kzalloc(sizeof(struct win2030_noc_error), GFP_ATOMIC);
 	if (!noc_err)
 		return -ENOMEM;
 
 	noc_err->timestamp = get_jiffies_64();
-
-	// Read all error log registers first but do not print
+	dev_err(mydev, "Error interrupt happen!\n");
 	for (i = 0; i < noc_device->error_logger_cnt; i++) {
 		noc_err->err[noc_device->err_log_lut[i]] = ioread32(ERRLOG_0_ERRLOG0(base) +
 					noc_device->err_log_lut[i] * sizeof(u32));
+		dev_err(mydev, "ErrLog%d 0x%08x", noc_device->err_log_lut[i],
+			noc_err->err[noc_device->err_log_lut[i]]);
 	}
-
-	// Check if the high 24 bits of ErrLog1 equal 0x000028**, it means uart noc err report
-	if (noc_err->err[1] >> 8 == 0x28)  // Check if upper 24 bits equal 0x000028
-		use_dbg = true;
-
-	// Select printing function based on the flag
-	if (use_dbg) {
-		dev_dbg(mydev, "Error interrupt happen!\n");
-		for (i = 0; i < noc_device->error_logger_cnt; i++) {
-			dev_dbg(mydev, "ErrLog%d 0x%08x", noc_device->err_log_lut[i],
-				noc_err->err[noc_device->err_log_lut[i]]);
-		}
-	} else {
-		dev_err(mydev, "Error interrupt happen!\n");
-		for (i = 0; i < noc_device->error_logger_cnt; i++) {
-			dev_err(mydev, "ErrLog%d 0x%08x", noc_device->err_log_lut[i],
-				noc_err->err[noc_device->err_log_lut[i]]);
-		}
-	}
-
 	spin_lock_irqsave(&noc_device->lock, flags);
 	list_add_tail(&noc_err->link, &noc_device->err_queue);
 	spin_unlock_irqrestore(&noc_device->lock, flags);
 
 	noc_error_dump(buf, noc_device, noc_err);
-
-	// Select print level for final error dump based on the flag
-	if (use_dbg)
-		dev_dbg(mydev, "%s\n", buf);
-	else
-		dev_err(mydev, "%s\n", buf);
-
-	// stack_dump();
+	dev_err(mydev, "%s\n", buf);
+	//stack_dump();
 	iowrite32(1, ERRLOG_0_ERRCLR(base));
 	return 0;
 }
 
-
 static irqreturn_t win2030_noc_error_irq(int irq, void *dev)
 {
 	struct win2030_noc_device *noc_device = dev;
-- 
2.53.0

